### PR TITLE
St 1382 fixing existing hearings listings

### PR DESCRIPTION
**Pull request by osheaf01**
### Change description ###

Fixing Prod issue with missing existing hearing

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ST-1382

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

**Review by sjamihmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCase.java [hunk 1]
```diff
@@ -53,6 +54,9 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDet
 
         log.info("Migrating case data for case Id: {}", details.getId());
 
+        CaseData data = details.getData();
+        data.setCaseBundles(null);
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCase.java:57]: This is to be done by the RetiredFields class - not here

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/task/SystemFinalOrderOverdueTask.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        try {
+            final BoolQueryBuilder query =
+                boolQuery()
+                    .must(boolQuery()
+                        .must(matchQuery("reference", 1688978122333564L))
+                    )
+
+                    /*.must(
+                        boolQuery()
+                            .should(boolQuery().must(rangeQuery(String.format(DATA, PRONOUNCED_DATE)).lt(overdueDate)))
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/task/SystemFinalOrderOverdueTask.java]: Creation Date would be much more suitable as you want to migrate all cases created before the code that is causing the issue was merged

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/SystemMigrateOldHearing.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import uk.gov.hmcts.sptribs.ciccase.model.CaseData;
+import uk.gov.hmcts.sptribs.ciccase.model.State;
+import uk.gov.hmcts.sptribs.ciccase.model.UserRole;
+
+import static uk.gov.hmcts.sptribs.ciccase.model.UserRole.SYSTEMUPDATE;
+import static uk.gov.hmcts.sptribs.ciccase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class SystemMigrateOldHearing implements CCDConfig<CaseData, State, UserRole> {
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/SystemMigrateOldHearing.java:20]: If this is a system update event it should be in the system update folder/module

Comment (Shailesh-walke) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/SystemMigrateOldHearing.java]: updated

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCase.java [hunk 4]
```diff
@@ -21,40 +21,27 @@ public class SystemMigrateCase implements CCDConfig<CaseData, State, UserRole> {
...
-    }
-
     public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
                                                                        final CaseDetails<CaseData, State> beforeDetails) {
-
         log.info("Migrating case data for case Id: {}", details.getId());
 
         CaseData data = details.getData();
-        data.setCaseBundles(null);
+        hearingService.addListingIfExists(data);
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCase.java]: Refactored to SystemMigrateOldHearing

**Review by ATedy — APPROVED** [review]

**Pull request merged**